ark/schema/structure/prop.ts

```diff
@@ -143,6 +152,9 @@ export abstract class BaseProp<
 	hasDefault(): this is Optional.Node & { default: unknown } {
 		return "default" in this
 	}
+	hasDefaultFactory(): this is Optional.Node & { default: () => unknown } {
```

Comment:
We should just add another method like `getDefault` for this that will automatically invoke it if it's a function.